# HG changeset patch
# User mcimadamore
# Date 1249949533 -3600
# Node ID abe992640c5a766e54f8490cf1d85ff2aacbbb5d
# Parent  d5f6c475f475b905f04446ea54ff455cd9d067fe
6390045: Unexpected error "cannot access java.lang.Void" with '-target cldc1.0' with -source >=1.5
Summary: javac needs to synthetize a fake java.lang.Void symbol if one is not given on the classpath
Reviewed-by: jjg

diff -r d5f6c475f475 -r abe992640c5a src/share/classes/com/sun/tools/javac/code/Symtab.java
--- openjdk.orig/langtools/src/share/classes/com/sun/tools/javac/code/Symtab.java	Tue Aug 11 01:11:37 2009 +0100
+++ openjdk/langtools/src/share/classes/com/sun/tools/javac/code/Symtab.java	Tue Aug 11 01:12:13 2009 +0100
@@ -440,6 +440,7 @@
         synthesizeEmptyInterfaceIfMissing(serializableType);
         synthesizeBoxTypeIfMissing(doubleType);
         synthesizeBoxTypeIfMissing(floatType);
+        synthesizeBoxTypeIfMissing(voidType);
 
         // Enter a synthetic class that is used to mark internal
         // proprietary classes in ct.sym.  This class does not have a
diff -r d5f6c475f475 -r abe992640c5a test/tools/javac/6390045/T6390045a.java
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ openjdk/langtools/test/tools/javac/6390045/T6390045a.java	Tue Aug 11 01:12:13 2009 +0100
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2009 Sun Microsystems, Inc.  All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Sun Microsystems, Inc., 4150 Network Circle, Santa Clara,
+ * CA 95054 USA or visit www.sun.com if you need additional information or
+ * have any questions.
+ */
+
+/*
+ * @test
+ * @bug 6390045
+ * @summary Unexpected error "cannot access java.lang.Void" with '-target cldc1.0' with -source >=1.5
+ *
+ * @author mcimadamore
+ * @compile -XDfailcomplete=java.lang.Void T6390045a.java
+ */
+
+class T6390045a {
+    boolean b;
+    short s;
+    Object o;
+    Object p = b ? o : s;
+}
diff -r d5f6c475f475 -r abe992640c5a test/tools/javac/6390045/T6390045b.java
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ openjdk/langtools/test/tools/javac/6390045/T6390045b.java	Tue Aug 11 01:12:13 2009 +0100
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2009 Sun Microsystems, Inc.  All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Sun Microsystems, Inc., 4150 Network Circle, Santa Clara,
+ * CA 95054 USA or visit www.sun.com if you need additional information or
+ * have any questions.
+ */
+
+/*
+ * @test
+ * @bug 6390045
+ * @summary Unexpected error "cannot access java.lang.Void" with '-target cldc1.0' with -source >=1.5
+ *
+ * @author mcimadamore
+ * @compile -XDfailcomplete=java.lang.Void T6390045b.java
+ */
+
+class T6390045b {
+    short s;
+    Object o;
+    Object p = choose(o, s);
+    <T> T choose(T t1, T t2) { return t1; }
+}
